### PR TITLE
Add a iiif v2 probe service

### LINKED ISSUE #1061: PDF Viewer Auth Flow: Implement IIIF Probe Service
Part of sul-dlss/sul-embed#1757 - auth flow for PDF viewer

~~Investigate~~ Implement IIIF Probe Service (https://iiif.io/api/auth/2.0/#probe-service) instead of creating additional auth endpoint for files (see bullet points 1 and 2 here: https://docs.google.com/document/d/17nhSWDp28CocKsBdcZvtqAOMiNbLnVspBRXENSr6m84/edit#heading=h.fknftcv8cgfk)

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #1061

Creates a new IIIF Probe Service (see https://github.com/sul-dlss/stacks/issues/1061) so that we can check auth for things other than media files and show appropriate messaging to users.

Eventually refactor media to use this new endpoint instead of its custom auth endpoint.

### CODE REVIEW

File: app/services/media_authentication_json.rb [hunk 1]
```diff
@@ -22,17 +22,6 @@ def initialize(auth_url)
...
-    # labels from https://consul.stanford.edu/display/chimera/Rights+Metadata+Locations
-    LOCATION_LABELS = {
-      'spec' => 'Special Collections reading room',
-      'music' => 'Music Library - main area',
-      'ars' => 'Archive of Recorded Sound listening room',
-      'art' => 'Art Library',
-      'hoover' => 'Hoover Library',
-      'm&m' => 'Media & Microtext'
-    }.freeze
-
```
Comment (peetucket) [inline app/services/media_authentication_json.rb:35]: move to settings to avoid duplicating in our new service

**Comment by peetucket** [comment]
More tests added; URI parsing changed since resource URIs do not encode paths.  See for example: https://sul-purl-stage.stanford.edu/zh939cr1328  which has files like this: https://sul-stacks-stage.stanford.edu/file/druid:zh939cr1328/105/105.txt

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -15,12 +15,12 @@ Layout/MultilineMethodCallBraceLayout:
     - 'spec/routing/file_routing_spec.rb'
     - 'spec/routing/media_routing_spec.rb'
 
-# Offense count: 8
+# Offense count: 9
 # Configuration parameters: AllowedMethods, AllowedPatterns, CountRepeatedAttributes.
 Metrics/AbcSize:
-  Max: 22
+  Max: 37
```
Comment (jcoyne) [inline .rubocop_todo.yml:21]: Can we avoid this change and instead inline exceptions so we can see where they're at?

File: app/controllers/iiif/auth/v2/probe_service_controller.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          render json: response
+        end
+
+        private
+
+        # add details to response for when access is denied
+        def add_detail(file)
+          detail = {}
+          if file.stanford_restricted? && !file.embargoed?
+            detail[:heading] = { en: ["Stanford-affiliated? Login to play"] }
```
Comment (jcoyne) [inline app/controllers/iiif/auth/v2/probe_service_controller.rb:36]: Can we put the headings in I18n?

File: config/settings.yml [hunk 4]
```diff
@@ -38,6 +38,14 @@ user:
     - sulair:proxy-access
     - lane:proxy-access
   locations:
+    # Codes from https://github.com/sul-dlss/cocina-models/blob/8fc7b5b9b0e3592a5c81f4c0e4ebff5c926669c6/openapi.yml#L1330-L1339
+    # labels from https://consul.stanford.edu/display/chimera/Rights+Metadata+Locations
+    labels: { spec: 'Special Collections reading room',
```
Comment (jcoyne) [inline config/settings.yml:43]: Typically we don't put curly braces in yaml

File: .rubocop_todo.yml (hunk 2 above)
Comment (peetucket) [reply .rubocop_todo.yml]: Restored todo and inlined exception

File: app/controllers/iiif/auth/v2/probe_service_controller.rb (hunk 3 above)
Comment (peetucket) [reply app/controllers/iiif/auth/v2/probe_service_controller.rb]: Done.

File: config/settings.yml (hunk 4 above)
Comment (peetucket) [reply config/settings.yml]: Fixed

File: config/settings.yml [hunk 5]
```diff
@@ -38,6 +38,15 @@ user:
...
   locations:
+    # Codes from https://github.com/sul-dlss/cocina-models/blob/8fc7b5b9b0e3592a5c81f4c0e4ebff5c926669c6/openapi.yml#L1330-L1339
+    # labels from https://consul.stanford.edu/display/chimera/Rights+Metadata+Locations
+    labels:
+      spec: 'Special Collections reading room'
+      music: 'Music Library - main area'
+      ars: 'Archive of Recorded Sound listening room'
+      art: 'Art Library'
+      hoover: 'Hoover Library'
+      m&m: 'Media & Microtext'
```
Comment (mjgiarlo) [inline config/settings.yml:49]: YAML question: does this hash key come through OK? Might be worth tweaking a test to make sure, or just running a manual test?

Comment (mjgiarlo) [reply config/settings.yml:49]: I'll merge this and run a test myself, nevermind!

**Pull request merged**